microsoft-edge/webview2/get-started/hololens2.md

```diff
@@ -0,0 +1,623 @@
+
+```C#
+public interface IWithMouseEvents : IWithInputEvents
+{
+    void MouseEvent(WebViewMouseEventData mouseEvent);
+}
+```
+
+WebView2 is unaware of Unity's input system and likely has a different coordinate system than your Unity scene. As a result, when there is a pointer-down event, its coordinates must be translated into the coordinate system of the WebView2 control. Additionally, the pointer-down event needs to be converted into an appropriate `WebViewMouseEventData` event type.
+
+Simple example:
+
+```C#
+using Microsoft.MixedReality.WebView;
+
+public class WebViewExample : MonoBehaviour, IPointerDownHandler
+{
+    // WebView setup steps skipped for brevity
+
+    public void OnPointerDown(PointerEventData eventData)
+    {
+        IWithMouseEvents mouseEventsWebView = webView as IWithMouseEvents;
+
+        // Call hypothetical function which converts the event's x, y into the WebView2's coordinate space.
+        var hitCoord = ConvertToWebViewSpace(eventData.position.x, eventData.position.y);
+
+        WebViewMouseEventData mouseEvent = new WebViewMouseEventData
+        {
+            X = hitCoord.x,
+            Y = hitCoord.y,
+            Type = PointerEvent.PointerDown,
+            Button = PointerButton.Left,
+            TertiaryAxisDeviceType = WebViewMouseEventData.TertiaryAxisDevice.PointingDevice
+        };
+
+        // Propagate the event to the WebView plugin.
+        mouseEventsWebView.MouseEvent(mouseEvent);
+    }
+}
+```
+
+In the above example, pointer-down events are converted into `WebViewMouseEventData` objects and forwarded to the WebView plugin for Unity. It is essentially converted into a mouse-down event. In order to effect mouse click events, pointer-up events would need to be handled in a similar fashion.
+
+In the example above, `ConvertToWebViewSpace` is intentionally not implemented.
+
+
+<!-- ------------------------------ -->
+#### Performance optimization
+
+Optimizing the performance of WebView2 in your HoloLens 2 Unity app is crucial for a smooth user experience. Here are some recommendations:
+
+* **Limit the number of WebView2 instances**: We suggest using only one instance of WebView2 within a Unity app. Reuse the same instance or tear down and create a new one as needed. Keep in mind that removing the WebView prefab from the scene may not destroy the underlying WebView2 instance. You must call the `Destroy()` method on the game object to destroy it properly.
+
+* **Use TrySuspendAsync API**: When the WebView2 is not being presented to the user, use the `TrySuspendAsync` API to reduce memory usage. You will need to get the native WebView2 instance, to call this API (⚠️ this is TBD, and we will update this section<!--todo--> accordingly ⚠️).
```

Comment:
link to this api ref?  https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.trysuspendasync?view=webview2-dotnet-1.0.1722.45

if so, copy/paste tab-set tagging including terminator `---`; find Tab in this file, eg:

```
##### [.NET/C#](#tab/dotnetcsharp)

* [CoreWebView2.SetVirtualHostNameToFolderMapping Method](/dotnet/api/microsoft.web.webview2.core.corewebview2.setvirtualhostnametofoldermapping)

##### [WinRT/C#](#tab/winrtcsharp)

* [CoreWebView2.SetVirtualHostNameToFolderMapping Method](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2#setvirtualhostnametofoldermapping)

##### [Win32/C++](#tab/win32cpp)

* [ICoreWebView2_3::SetVirtualHostNameToFolderMapping method](/microsoft-edge/webview2/reference/win32/icorewebview2_3#setvirtualhostnametofoldermapping)

---

```
If we want such links, i can implement or finalize that